src/core/evaluator.js

```diff
@@ -2007,9 +2007,8 @@ class PartialEvaluator {
                   localColorSpaceCache,
                 })
                 .then(function (colorSpace) {
-                  if (colorSpace) {
-                    stateManager.state.fillColorSpace = colorSpace;
-                  }
+                  stateManager.state.fillColorSpace =
+                    colorSpace || ColorSpace.singletons.gray;
```

Comment:
If this is necessary, should we not make the same sort of change below as well if nothing else than for consistency?
https://github.com/mozilla/pdf.js/pull/18467/files#diff-cf40014d9c7d352f3ae212e7fffe5386decf96a10e53066a4d900c3eb8fc559eR2035-R2037

Reply:
I wondered the same and I don't know. I'm even not sure that fallbacking on gray is a good choice in general, at least it fixes the 2 test failures in the previous run.

Reply:
Well, as mentioned in https://github.com/mozilla/pdf.js/issues/18466#issuecomment-2241107239 it seems that `DeviceGray` should be the default value.

So, for consistency, we do need to update the `OPS.setStrokeColorSpace` case below as well.